### PR TITLE
fix: standardize AgentOS media validation

### LINKED ISSUE #7311: [Technical] Standardize Media Validation and Unify AgentOS Routers
**Context**
Currently, the AgentOS routers (Agents and Teams) handle file uploads using manual, hardcoded MIME type checks. This leads to:
1. Inconsistent media handling across different routers.
2. Fragile validation that misses cases like case-insensitive MIME types (e.g., `IMAGE/PNG`) or special characters in filenames.
3. Code duplication for processing `Image`, `Video`, and `File` media types.

**Solution**
Unify media processing by:
- Moving validation logic into the `Media` base classes (`Image`, `Video`, `File`).
- Centralizing file processing in the routers to use these standardized methods.
- Adding robust normalization for MIME types and special characters in filenames.
- Adding comprehensive unit tests for these edge cases.

### PR DESCRIPTION
**Pull request by weichu2002**
## Summary

Fixes #7311.

Standardizes AgentOS upload MIME handling in the shared media and router utility layers. Agent and Team routers already use the same `classify_upload_file` / processing entry points; this change removes the remaining case-sensitive MIME behavior there and in `File` validation.

## Type of change

- [x] Bug fix

---

## Checklist

- [x] Code complies with style guidelines
- [ ] Ran format/validation scripts (`./scripts/format.sh` and `./scripts/validate.sh`) — local `.venv` is not configured
- [x] Self-review completed
- [x] Documentation updated (comments, docstrings)
- [ ] Tested in clean environment — not available locally
- [x] Tests added/updated

### Duplicate and AI-Generated PR Check

- [x] I have searched existing open pull requests and confirmed that no other PR already addresses this issue
- [ ] If a similar PR exists, I have explained below why this PR is a better approach
- [x] Check if this PR was entirely AI-generated — AI-assisted; reviewed before submission

---

### Root cause and consistency

The latest main already centralizes Agent and Team Router upload classification in `agno.os.utils`, including Excel and MSG MIME support. The remaining inconsistency was strict MIME comparison in the shared upload functions and `File` validator: mixed-case or padded values such as `IMAGE/PNG` and ` Application/PDF ` were rejected.

### Shared validation

`normalize_mime_type()` is shared from the media module and trims/lowercases MIME values. The shared upload classifier, Image/Video construction, document resolution, and `File` validator reuse it. Existing known MIME continues to take precedence over filenames; ambiguous MIME may use the extension fallback as before, and an extensionless octet-stream upload remains unsupported.

### Tests

Added boundary coverage for image, video, and document uploads with mixed-case MIME, Chinese/space/parenthesis/multi-dot filenames, no extension, and MIME/extension mismatch. Because both routers call the same utilities, this validates their shared behavior.

### Commands run

- `python -m py_compile libs/agno/agno/media.py libs/agno/agno/os/utils.py libs/agno/tests/unit/os/test_utils.py` — passed
- `git diff --check` — passed
- `pytest`, `./scripts/format.sh`, and `./scripts/validate.sh` — not run because local `.venv` is not configured.

### CODE REVIEW

File: libs/agno/agno/os/utils.py [hunk 1]
```diff
@@ -677,7 +677,7 @@ def classify_upload_file(file: UploadFile) -> Optional[str]:
     to be useful (common for `.md` and `.pptx` uploaded from browsers), falls back to the
     filename extension. Returns None if the file type is not supported.
     """
-    content_type = file.content_type
+    content_type = normalize_mime_type(file.content_type)
```
[the remark this answers is not in the document]
Comment (weichu2002) [reply libs/agno/agno/os/utils.py:680]: Thanks for catching this. `classify_upload_file()` normalized the MIME for routing,
but `process_audio()` still retained the original value. I’ve updated the shared
audio processing path to store the normalized MIME type and added a regression
test covering padded, uppercase audio MIME values.
